src/components/Exercises.js

```diff
@@ -69,18 +70,18 @@ const Exercises = ()=>{
         <button onClick={navigateExerciseForm}>Add New Exercise</button>
         <ul>
           {exercisesData.exercises.map(curr=>{
-            let currWorkout 
-            exercisesData.workouts.forEach((item)=>{
-              if(item._id === curr.workout){
-                currWorkout = item
-              }
-            })
+            // let currWorkout 
+            // exercisesData.workouts.forEach((item)=>{
+            //   if(item._id === curr.workout){
+            //     currWorkout = item
+            //   }
+            // })
         
             return(
               <li >
                 <h3>Title: {curr.title}</h3>
                 <p>Category: {curr.category}</p>
-                <p>Workout: <button onClick={()=>navigateWorkout(currWorkout._id)}>{currWorkout.title}</button></p>
+                <p>Workout: <button onClick={()=>navigateWorkout(curr._id)}>{curr.title}</button></p>
```

Comment:
Again, not explicitly needed here, but for future reference, writing an arrow function here creates a new function on every render which, in JS, means a new object each time, which could mean higher memory consumption. Most likely not in this scenario, but just something to be aware of.

React has something called `useCallback`, that is used kind of like `useEffect`:
```
const handleClick = useCallback(() => {
    navigateWorkout(curr._id);
  }, [curr._id, navigateWorkout]);
```

And then later, on this line, it would just be:
```
<p>Workout: <button onClick={handleClick}>{curr.title}</button></p>
```

Reply:
I didnt know that!

Reply:
Should I take it a step further and remove the naviagte workout fiunction and just navigate in the handleWorkoutClick function or keep them separated? I figure since its fairly clear what navigate is doing and it isnt used elsewhere that keeping them separate doesnt serve a huge purpose

Reply:
Also it keeps telling me that curr is not defined. I assumed that it would be able to pull it from where it is called. chaytgpt is telling me to move the functions inside of the map which to me seems a little messy. What do you think?

Reply:
<div>
        <h1>Your Exercises</h1>
        <button onClick={navigateExerciseForm}>Add New Exercise</button>
        <ul>
          {exercisesData.exercises.map(curr => {
            const handleWorkoutClick = () => {
              navigateWorkout(curr._id);
            };

            const handleButtonClick = () => {
              navigateStats(curr.id);
            };

            return (
              <li key={curr._id}>
                <h3>Title: {curr.title}</h3>
                <p>Category: {curr.category}</p>
                <p>Workout: <button onClick={handleWorkoutClick}>{curr.title}</button></p>
                <button onClick={handleButtonClick}>See Stats</button>
              </li> // TO-DO: add functionality to open to exercise page with stats.
            )
          })}
        </ul>``

Reply:
I think removing the navigate function is fine. In your code above, if I am reading it correctly, it looks like you are defining the `handleButtonClick` inside of the array map, which kind of brings us back to where we were before, which was creating a new function on each render.

If that will be the case, I think something more along the lines of what you were doing before is more readable and maintainable. Something like this:
```
{exercisesData.exercises.map(curr => (
    <li key={curr._id}>
        <h3>Title: {curr.title}</h3>
        <p>Category: {curr.category}</p>
        <p>Workout: <button onClick={() => navigateWorkout(curr._id)}>{curr.title}</button></p>
        <button onClick={() => navigateStats(curr._id)}>See Stats</button>
    </li>
))}
```

Reply:
Why do you think its not being able to access curr when I do it with use callback?

Reply:
but navigate workout does?